src/main/scala/ai/privado/utility/Utilities.scala

```diff
@@ -609,4 +608,18 @@ object Utilities {
       }
     })
   }
+
+  def addOriginalSourceEdgeAndTag(
+    builder: BatchedUpdate.DiffGraphBuilder,
+    impactedObject: AstNode,
+    originalSourceNode: AstNode,
+    ruleCache: RuleCache,
+    randomSlug: String
+  ): Unit = {
+    builder.addEdge(impactedObject, originalSourceNode, EdgeTypes.ORIGINAL_SOURCE)
+    storeForTag(builder, originalSourceNode, ruleCache)(
```

Comment:
Didn't understood the use of adding tag to the original node here?

Reply:
This is just for tracking, we'll know if there's a derived source coming out from that variable.